editor.planx.uk/src/@planx/components/PlanningConstraints/Modal.tsx

```diff
@@ -0,0 +1,247 @@
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import Divider from "@mui/material/Divider";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import visuallyHidden from "@mui/utils/visuallyHidden";
+import { Constraint, Metadata } from "@opensystemslab/planx-core/types";
+import omit from "lodash/omit";
+import React, { useState } from "react";
+import InputLabel from "ui/public/InputLabel";
+import ChecklistItem from "ui/shared/ChecklistItem";
+import ErrorWrapper from "ui/shared/ErrorWrapper";
+import Input from "ui/shared/Input";
+
+import { formatEntityName } from "./List";
+import { InaccurateConstraints } from "./Public";
+
+interface OverrideEntitiesModalProps {
+  showModal: boolean;
+  setShowModal: (value: React.SetStateAction<boolean>) => void;
+  fn: Constraint["fn"];
+  entities: Constraint["data"] | null;
+  metadata?: Metadata;
+  inaccurateConstraints: InaccurateConstraints;
+  setInaccurateConstraints: (
+    value: React.SetStateAction<InaccurateConstraints>,
+  ) => void;
+}
+
+const ERROR_MESSAGES = {
+  checklist: "Select at least one option",
+  input: "Enter a value",
+};
+
+export const OverrideEntitiesModal = ({
+  showModal,
+  setShowModal,
+  fn,
+  entities,
+  metadata,
+  inaccurateConstraints,
+  setInaccurateConstraints,
+}: OverrideEntitiesModalProps) => {
+  const [checkedOptions, setCheckedOptions] = useState<string[] | undefined>(
+    inaccurateConstraints?.[fn]?.["entities"],
+  );
+  const [showChecklistError, setShowChecklistError] = useState<boolean>(false);
+  const [textInput, setTextInput] = useState<string | undefined>(
+    inaccurateConstraints?.[fn]?.["reason"],
+  );
+  const [showInputError, setShowInputError] = useState<boolean>(false);
+
+  const title = `Which ${
+    metadata?.plural?.toLowerCase() || "entities"
+  } are inaccurate?`;
+
+  const closeModal = (_event: any, reason?: string) => {
+    if (reason && reason == "backdropClick") {
+      return;
+    }
+
+    // Clear any non-submitted inputs on cancel & sync parent state
+    setCheckedOptions(undefined);
+    setTextInput(undefined);
+    const newInaccurateConstraints = omit(inaccurateConstraints, fn);
+    setInaccurateConstraints(newInaccurateConstraints);
+
+    // Close modal
+    setShowModal(false);
+  };
+
+  const changeCheckbox =
+    (id: string) =>
+    (_checked: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined) => {
+      let newCheckedIds;
+
+      if (checkedOptions?.includes(id)) {
+        newCheckedIds = checkedOptions.filter((e) => e !== id);
+      } else {
+        newCheckedIds = [...(checkedOptions || []), id];
+      }
+
+      if (newCheckedIds.length > 0) {
+        setShowChecklistError(false);
+      }
+      setCheckedOptions(newCheckedIds);
+    };
+
+  const changeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value.length > 1) {
```

Comment:
```suggestion
    if (e.target.value.length) {
```

Little bug here - currently the error state persists if there's a single character submitted, once the error state is already triggered

![image](https://github.com/user-attachments/assets/2b6e754e-3c1a-459f-a358-d9a686ec6c4e)

Reply:
Now resolved!